src/cairoUtilFuncGen/memory/arrayLiteral.ts

```diff
@@ -108,41 +157,148 @@ export class MemoryArrayLiteralGen extends StringIndexedFuncGen {
     return funcDef;
   }
 
-  private getOrCreate(type: TypeNode, size: number, dynamic: boolean): GeneratedFunctionInfo {
+  private getOrCreate(
+    type: TypeNode,
+    size: number,
+    dynamic: boolean,
+    isUserDefined: boolean,
+    isMatrix: boolean,
+    is3dMatrix: boolean,
```

Comment:
Hi Echo, I think this approach is not considering arrays of 3 or more dimentions(eg: `uint[a][b][c][d]` being `a`,`b`,`c` and `d` constants) at least is not working for me. Should it be working?

Reply:
okay @AlejandroLabourdette , i'm a little busy with this issue https://github.com/NethermindEth/warp/issues/987 , when i finish I will update this PR to the Cairo-1.0 syntax and fix the problem that you pointed out. Thank you